### PR TITLE
FIX TinyMCE Insert link from Files

### PR DESCRIPTION
**Pull request by sabina-talipova**
## Description
- It looks like using `$` conflicts with another library, so it was replaced with `jQuery`;
- Fix existing `getSelection` function issue. ~~[See](https://github.com/silverstripe/silverstripe-admin/issues/1548)~~   
## Parent issue
- https://github.com/silverstripe/silverstripe-cms/issues/2915

### CODE REVIEW

File: client/src/entwine/TinyMCE_ssmedia.js [hunk 1]
```diff
@@ -230,13 +230,7 @@ jQuery.entwine('ss', ($) => {
       const { url, ...attrs } = this.getOriginalAttributes();
       const fileSelected = attrs.hasOwnProperty('ID') && attrs.ID !== null;
       const folderId = this.getFolderId();
-      const editor = this.getElement().getEditor();
-      const selection = editor.getInstance().selection;
-      const selectionContent = editor.getSelection();
-      const tagName = selection.getNode().tagName;
-      // Unsupported media insertion will use insert link form instead
```
Comment (ChloeSartorelli) [inline client/src/entwine/TinyMCE_ssmedia.js]: We need to keep this comment - it gives context about why we even have a `requireLinkText` variable in a js file that doesn't seem to be related to links

File: client/src/entwine/TinyMCE_ssmedia.js [hunk 2]
```diff
@@ -230,13 +230,7 @@ jQuery.entwine('ss', ($) => {
...
-      // treat image tag selection as blank content
-      const requireLinkText = tagName !== 'A' && (tagName === 'IMG' || selectionContent.trim() === '');
+      const requireLinkText = this.getRequireLinkText();
```
Comment (ChloeSartorelli) [inline client/src/entwine/TinyMCE_ssmedia.js]: This comment, and the `tagName === 'IMG'` portion of this logic, are both important.

If you add an image, and then try to replace that image with a pdf, the "Link text" field should be present - but this PR removes that.

Before this PR:
[Screencast from 25-01-24 15:40:50.webm](https://github.com/silverstripe/silverstripe-asset-admin/assets/36352093/ca0ac7a1-f018-4aa2-845d-1a5a2cf49007)

With this PR:
[Screencast from 25-01-24 15:43:05.webm](https://github.com/silverstripe/silverstripe-asset-admin/assets/36352093/1e0c7db3-bdfe-4812-b382-c60bfd168135)




Also note that the `getRequireLinkText()` function is defined for elements with the CSS selector `'.insert-link__dialog-wrapper'` in https://github.com/silverstripe/silverstripe-admin/pull/1579/. If the function exists to be called for this element, it's purely coincidence and shouldn't be relied upon.

**Review by ChloeSartorelli — CHANGES_REQUESTED** [review]
The fix for the bug reported in the issue is fine - but the stuff you've added for the required link text doesn't work as expected. Details below.

File: client/src/entwine/TinyMCE_ssmedia.js (hunk 1 above)
Comment (sabina-talipova) [reply client/src/entwine/TinyMCE_ssmedia.js]: Updated

File: client/src/entwine/TinyMCE_ssmedia.js (hunk 2 above)
Comment (sabina-talipova) [reply client/src/entwine/TinyMCE_ssmedia.js]: Updated. Cancel changes.

**Review by ChloeSartorelli — APPROVED** [review]
LGTM
